src/Facets/HopFacetPacked.sol

```diff
@@ -0,0 +1,435 @@
+            bonderFee: getCalldataValue(52, 16), // bytes16(uint128) > uint256
+            amountOutMin: getCalldataValue(68, 16), // bytes16(uint128) > uint256
+            destinationAmountOutMin: getCalldataValue(84, 16), // bytes16(uint128) > uint256
+            hopBridge: address(uint160(getCalldataValue(100, 20))) // bytes20 > address
+            // => total calldata length required: 120
+        });
+    }
+
+    /// @notice Bridges Native tokens via Hop Protocol from L2
+    /// @param transactionId Custom transaction ID for tracking
+    /// @param integrator LI.FI partner name
+    /// @param receiver Receiving wallet address
+    /// @param destinationChainId Receiving chain
+    /// @param bonderFee Fees payed to hop bonder
+    /// @param amountOutMin Source swap minimal accepted amount
+    /// @param destinationAmountOutMin Destination swap minimal accepted amount
+    /// @param hopBridge Address of the Hop L2_AmmWrapper
+    function startBridgeTokensViaHopL2NativeMin(
+        bytes32 transactionId,
+        string calldata integrator,
+        address receiver,
+        uint256 destinationChainId,
+        uint256 bonderFee,
+        uint256 amountOutMin,
+        uint256 destinationAmountOutMin,
+        address hopBridge
+    ) external payable {
+        _startBridgeTokensViaHopL2Native(
+            transactionId,
+            integrator,
+            receiver,
+            destinationChainId,
+            bonderFee,
+            amountOutMin,
+            destinationAmountOutMin,
+            hopBridge
+        );
+    }
+
+    /// @notice Bridges Native tokens via Hop Protocol from L2
+    /// @param transactionId Custom transaction ID for tracking
+    /// @param integrator LI.FI partner name
+    /// @param receiver Receiving wallet address
+    /// @param destinationChainId Receiving chain
+    /// @param bonderFee Fees payed to hop bonder
+    /// @param amountOutMin Source swap minimal accepted amount
+    /// @param destinationAmountOutMin Destination swap minimal accepted amount
+    /// @param hopBridge Address of the Hop L2_AmmWrapper
+    function encoder_startBridgeTokensViaHopL2NativePacked(
+        bytes32 transactionId,
+        string calldata integrator,
+        address receiver,
+        uint256 destinationChainId,
+        uint256 bonderFee,
+        uint256 amountOutMin,
+        uint256 destinationAmountOutMin,
+        address hopBridge
+    ) external pure returns (bytes memory) {
+        return bytes.concat(
+            abi.encodeWithSignature("startBridgeTokensViaHopL2NativePacked()"),
+            bytes8(transactionId),
+            bytes16(bytes(integrator)),
+            bytes20(receiver),
+            bytes4(uint32(destinationChainId)),
+            bytes16(uint128(bonderFee)),
+            bytes16(uint128(amountOutMin)),
+            bytes16(uint128(destinationAmountOutMin)),
+            bytes20(hopBridge)
+        );
```

Comment:
Perhaps check here the values fit within the destination bytes,
When this call is done off chain gas cost wouldn't be an issue